ovh/resource_ovh_cloud_kubernetes_node_pool.go

```diff
@@ -0,0 +1,392 @@
+		diags = append(diags, diag.Diagnostic{
+			Severity:      diag.Error,
+			Summary:       "Failed to update node pool",
+			Detail:        err.Error(),
+			AttributePath: nil,
+		})
+		return
+	}
+
+	log.Printf("[DEBUG] Waiting for Node %s:", poolId)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"INSTALLING", "UPDATING", "REDEPLOYING", "RESIZING"},
+		Target:     []string{"READY"},
+		Refresh:    waitForCloudKubernetesNodePoolActive(config.OVHClient, projectId, clusterId, poolId),
+		Timeout:    45 * time.Minute,
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		{
+			diags = append(diags, diag.Diagnostic{
+				Severity:      diag.Error,
+				Summary:       "Failed to get node pool ready",
+				Detail:        err.Error(),
+				AttributePath: nil,
+			})
+			return
+		}
+	}
+
+	err = resourceCloudKubernetesNodePoolRead(d, meta)
+
+	if err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity:      diag.Error,
+			Summary:       "Failed to read node pool",
+			Detail:        err.Error(),
+			AttributePath: nil,
+		})
+		return
+	}
+
+	d.Partial(false)
+
+	return nil
+
+}
+
+func resourceCloudKubernetesNodePoolCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
+	config := meta.(*Config)
+
+	projectId := d.Get("project_id").(string)
+	clusterId := d.Get("cluster_id").(string)
+	params := &CloudKubernetesNodePoolCreationRequest{
+		Name:          d.Get("name").(string),
+		FlavorName:    d.Get("flavor").(string),
+		DesiredNodes:  d.Get("desired_nodes").(int),
+		MaxNodes:      d.Get("max_nodes").(int),
+		MinNodes:      d.Get("min_nodes").(int),
+		MonthlyBilled: d.Get("monthly_billed").(bool),
+	}
+
+	r := &CloudKubernetesNodePoolResponse{}
+
+	log.Printf("[DEBUG] Will create public cloud kubernetes cluster: %s", params)
+
+	d.Partial(true)
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/kube/%s/nodepool", projectId, clusterId)
+
+	err := config.OVHClient.Post(endpoint, params, r)
+	if err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity:      diag.Error,
+			Summary:       "Failed to create node pool",
+			Detail:        err.Error(),
+			AttributePath: nil,
+		})
+		return
+	}
+
+	d.SetId(r.Id)
+
+	// This is a fix for a weird bug where the cluster is not immediately available on API
+	bugFixWait := &resource.StateChangeConf{
+		Pending:    []string{"NOT_FOUND"},
+		Target:     []string{"FOUND"},
+		Refresh:    waitForCloudKubernetesNodePoolToBeReal(config.OVHClient, projectId, clusterId, r.Id),
+		Timeout:    2 * time.Minute,
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	log.Printf("[DEBUG] Waiting for Node %s:", r)
+	_, err = bugFixWait.WaitForStateContext(ctx)
+
+	log.Printf("[DEBUG] Waiting for Node %s:", r)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"INSTALLING", "UPDATING", "REDEPLOYING", "RESIZING"},
```

Comment:
Same suggestion that for cluster statuses: have an enum listing all node statuses would be great.

Reply:
we dont use enums for statechangeconf elsewhere.
we could introduce such mechanism is a separated PR for a refactoring

Reply:
IMO, this is not a blocker (I approved the PR).
The suggestion was to implements this kind of enum: https://github.com/ovh/terraform-provider-ovh/pull/160#discussion_r527134576